### PR TITLE
Mixpanel service

### PR DESCRIPTION
**Pull request by eishay**
@andrewconner @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/dev.html [hunk 1]
```diff
@@ -27,6 +30,11 @@
         d.getElementsByTagName('BODY')[0].appendChild(s);
       }
     }(document, 'script');
+
+    (function(e,b){if(!b.__SV){var a,f,i,g;window.mixpanel=b;a=e.createElement("script");a.type="text/javascript";a.async=!0;a.src='//cdn.mxpnl.com/libs/mixpanel-2.2.min.js';f=e.getElementsByTagName("script")[0];f.parentNode.insertBefore(a,f);b._i=[];b.init=function(a,e,d){function f(b,h){var a=h.split(".");2==a.length&&(b=b[a[0]],h=a[1]);b[h]=function(){b.push([h].concat(Array.prototype.slice.call(arguments,0)))}}var c=b;"undefined"!==typeof d?c=b[d]=[]:d="mixpanel";c.people=c.people||[];c.toString=function(b){var a="mixpanel";"mixpanel"!==d&&(a+="."+d);b||(a+=" (stub)");return a};c.people.toString=function(){return c.toString(1)+".people (stub)"};i="disable track track_pageview track_links track_forms register register_once alias unregister identify name_tag set_config people.set people.set_once people.increment people.append people.track_charge people.clear_charges people.delete_user".split(" ");for(g=0;g<i.length;g++)f(c,i[g]);b._i.push([a,e,d])};b.__SV=1.2}})(document,window.mixpanel||[]);
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/dev.html:29]: In the head and bottom of body?

File: kifi-backend/modules/shoebox/angular/index.html [hunk 2]
```diff
@@ -25,6 +25,15 @@
     <section class="kf-col kf-main" ng-include="'layout/main/main.tpl.html'"></section>
     <aside class="kf-col kf-col-right" ng-include="'layout/rightCol/rightCol.tpl.html'"></aside>
   </div>
+  <script>
+    console.log("mixpanel setup")
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/index.html]: `console.log`

File: kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        if (locations[loc].test(path)) {
+          return loc;
+        }
+      }
+      return path;
+    }
+
+    function getUserStatus() {
+      var userStatus = 'standard';
+      if (profileService.me && profileService.me.experiments) {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: `if (profileService && profileService.me && profileService.me.experiments)`

File: kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            userStatus: getUserStatus()
+          });
+        } finally {
+          if (!oldId) {
+            mixpanel.identify(oldId);
+          }
+        }
+      } else {
+        identifiedViewEventQueue.push(path);
+        if (profileService.me && profileService.me.id) {
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: `if (profileService && profileService.me && profileService.me.id)`

File: kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        origin: $window.location.origin
+      });
+    }
+
+    angulartics.waitForVendorApi('mixpanel', 5000, function (/*mixpanel*/) {
+      $analyticsProvider.registerPageTrack(function (path) {
+        if (profileService && $window) {
+          var mixpanel = $window.mixpanel;
+          var normalizedPath = getLocation(path);
+          registerPageTrackForVisitor(mixpanel, normalizedPath, $window);
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: `$window` isn't used by this function

File: kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+          registerPageTrackForUser(mixpanel, normalizedPath, $window, profileService);
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: Looks like it should be `registerPageTrackForUser(mixpanel, normalizedPath, $window.location.origin);`

File: kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js (hunk 3 above)
Comment (eishay) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: you can't get to this line without profileService (there's a safeguard earlier for that). doesn't hurt to add for explicitness.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/common/services/mixpanelService.js]: Ah, I see. Yeah the global state messes me up :)

**Comment by eishay** [comment]
fixed

**Pull request merged**